src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CSharpInstructionDecoder.cs

```diff
@@ -105,7 +128,18 @@ internal override PEMethodSymbol GetMethod(DkmClrInstructionAddress instructionA
                 compilation = dataItem.Compilation;
             }
 
-            return compilation.GetSourceMethod(moduleInstance.Mvid, instructionAddress.MethodId.Token);
+            return compilation;
+        }
+
+        internal override MethodSymbol GetMethod(CSharpCompilation compilation, DkmClrInstructionAddress instructionAddress)
+        {
+            return compilation.GetSourceMethod(instructionAddress.ModuleInstance.Mvid, instructionAddress.MethodId.Token);
+        }
+
+        internal override TypeNameDecoder<PEModuleSymbol, TypeSymbol> GetTypeNameDecoder(CSharpCompilation compilation, MethodSymbol method)
+        {
+            Debug.Assert(method.GetType() == typeof(PEMethodSymbol));
```

Comment:
Either (method is PEMethodSymbol) or remove the Debug.Assert since the next statement will assert the same thing.